dlt-connector/src/server/createServer.ts

```diff
@@ -0,0 +1,55 @@
+import 'reflect-metadata'
+
+import { ApolloServer } from '@apollo/server'
+import { expressMiddleware } from '@apollo/server/express4'
+import express, { Express } from 'express'
+
+// graphql
+import { schema } from '@/graphql/schema'
+
+import { logger as dltLogger } from './logger'
+import { Logger } from 'log4js'
+import cors from 'cors'
+import bodyParser from 'body-parser'
+
+type ServerDef = { apollo: ApolloServer; app: Express }
+
+interface MyContext {
+  token?: string
+}
+
+const createServer = async (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  // context: any = serverContext,
+  logger: Logger = dltLogger,
+  // localization: i18n.I18n = i18n,
+): Promise<ServerDef> => {
+  logger.addContext('user', 'unknown')
+  logger.debug('createServer...')
+
+  // Express Server
+  const app = express()
+
+  // Apollo Server
+  const apollo = new ApolloServer<MyContext>({
+    schema: await schema(),
+    introspection: true,
+    // context,
+    // plugins
+    logger,
+  })
+  await apollo.start()
+  app.use(
+    '/',
+    cors<cors.CorsRequest>(),
+    bodyParser.json(),
+    expressMiddleware(apollo, {
+      context: async ({ req }) => ({ token: req.headers.token }),
+    }),
+  )
```

Comment:
what's the background for your changed creation-logic, because it looks quite different than the backend createServer?

Reply:
I use the current Apollo Version 4, not the deprecated Apollo Version 2 which is used from the rest project.
https://www.apollographql.com/docs/apollo-server/v2
Apollo Server 2 is officially deprecated, with end-of-life scheduled for 22 October 2023. Additionally, certain features are end-of-life as of 31 December 2023

I have copied createServer.ts from federation and had to adjust it, because it don't longer work.

Reply:
very nice, than you could collect experiences with this new version, so that your knowledge could help us for upgrading, which will coming soon , i gues...